### ISSUE TITLE
Outdated non-POSIX FreeBSD API used by tests

### ISSUE THREAD

**Issue by ctrlcctrlv** [rust-lang/libc#938]
`setgrent` returned an int two years ago, but now (as of https://github.com/freebsd/freebsd/commit/786c3926a65879b5dd6c301ca9ccdfda486fd195 ) returns void. We're testing against the outdated API, which became an issue in #934 

Thanks to @gnzlbg for finding the issue.

**Comment by bdrewery** [comment]
Note that `setgrent` did not get a new symbol version in FreeBSD for the changed ABI.

**Comment by gnzlbg** [comment]
@asomers IIUC this function still returns an `int` in FreeBSD 11, that is, we can't change its prototype to specify that it does not return anything. 

In FreeBSD12, the return value is _invalid_, and creating an _invalid_ value is UB in Rust. I agree that, as long as FreeBSD library is always dynamically linked, that is, as long as LTO is not possible, this UB cannot break code that ignores the return value.

**Comment by dybucc** [comment]
@tgross35 Not sure if there's anything to fix here anymore. There's already
proper handling of `setgrent` across each of the modules handling different
FreeBSD versions.

**Comment by tgross35** [comment]
Indeed, looks like we're covered https://github.com/search?q=repo%3Arust-lang%2Flibc+setgrent+path%3A%2F%5Esrc%5C%2Funix%5C%2F%2F+path%3A%2F%5Esrc%5C%2Funix%5C%2Fbsd%5C%2Ffreebsdlike%5C%2F%2F+path%3A%2F%5Esrc%5C%2Funix%5C%2Fbsd%5C%2Ffreebsdlike%5C%2Ffreebsd%5C%2F%2F&type=code&p=1 thanks for pointing it out.
